src/software/thunderscope/tab.py

```diff
@@ -0,0 +1,165 @@
+from typing import Callable, Optional, Sequence, Any, Dict
+from software.thunderscope.constants import TabNames
+
+import PyQt6
+from PyQt6.QtWebEngineWidgets import QWebEngineView
+from pyqtgraph.Qt import QtCore
+from pyqtgraph.Qt.QtWidgets import *
+from pyqtgraph.dockarea import *
+
+
+class WidgetStretchData:
+    """
+    Data that describes how a widget should be sized
+    """
+
+    x: Optional[int]  # stretch in x direction
+    y: Optional[int]  # stretch in y direction
+
+    def __init__(self, x: int = None, y: int = None):
+        self.x = x
+        self.y = y
+
+
+class TScopeWidget:
+    """
+    Data that describes a widget in Thunderscope
+    """
+
+    name: str  # name of widget (must be unique)
+    widget: Any  # the widget object
+    anchor: Optional[str]  # name of widget to position this relative to
+    position: Optional[str]  # position relative to anchor to position this to
+    stretch: Optional[WidgetStretchData]  # how widget should be sized
+    no_refresh: Optional[bool]  # if this widget has a refresh function or not
+    in_window: Optional[bool]  # if this widget should be added in window or not
```

Comment:
so this is just for the performance widget. For some reason, if I add it like the other widgets, it opens in a new window. 

So for that widget specifically, I have to add it as `widget.win` instead of just `widget`

`new_dock.addWidget(new_widget.win if data.in_window else new_widget)`

Reply:
I see. could you add a comment explaining what you just told me to where the Performance Plotter `TScopeWidget` is created?